pg-extern-attr/src/lib.rs

```diff
@@ -118,6 +120,22 @@ fn sql_return_type(outputs: &syn::ReturnType) -> TokenStream {
     quote!(pg_extend::pg_type::PgType::from_rust::<#ty>().return_stmt())
 }
 
+/// Returns Rust code to figure out if the function takes optional arguments. Functions with
+/// non-optional arguments will be declared with the STRICT option. PostgreSQL behavior:
+///
+/// > If this parameter is specified, the function is not executed when there are null arguments;
+/// > instead a null result is assumed automatically.
+fn sql_function_options(arg_types: &[&Type]) -> TokenStream {
+    if arg_types.is_empty() {
+        return quote!("",);
+    }
+
+    let arg_type = arg_types[0];
+    quote!(
+        if <#arg_type>::is_option() { "" } else { " STRICT" },
```

Comment:
looks like you got all this in nicely!

if you look at https://github.com/bluejekyll/pg-extend-rs/pull/36/files#diff-1dcddc586c5309d52ec002c650051db4R29 you'll see a way of combining a bunch of the output strings together. Doing something like that, you could either stick all the args into a list, and then do an `any()` over the entire iterator, or you could just list out each line... Up to you.